hr_expense_advance_clearing/models/hr_expense_sheet.py

```diff
@@ -0,0 +1,114 @@
+# Copyright 2019 Kitti Upariphutthiphong <kittiu@ecosoft.co.th>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, fields, _
+from odoo.tools.safe_eval import safe_eval
+from odoo.tools import pycompat
+from odoo.exceptions import ValidationError
+
+
+class HrExpenseSheet(models.Model):
+    _inherit = 'hr.expense.sheet'
+
+    advance = fields.Boolean(
+        string='Employee Advance',
+        compute='_compute_advance',
+        store=True,
+    )
+    advance_sheet_id = fields.Many2one(
+        comodel_name='hr.expense.sheet',
+        string='Clear Advance',
+        domain="[('advance', '=', True), ('employee_id', '=', employee_id),"
+               " ('residual', '>', 0.0)]",
+        readonly=True,
+        states={'draft': [('readonly', False)],
+                'submit': [('readonly', False)],
+                'approve': [('readonly', False)]},
+        help="Show remaining advance of this employee",
+    )
+    residual = fields.Monetary(
+        string='Amount to clear',
+        compute='_compute_residual',
+        store=True,
+        help="Amount to clear of this expense sheet in company currency",
+    )
+    advance_sheet_residual = fields.Monetary(
+        string='Advance Remaining',
+        related='advance_sheet_id.residual',
+        store=True,
+        help="Remaining amount to clear the selected advance sheet",
+    )
+    amount_payable = fields.Monetary(
+        string='Payable Amount',
+        compute='_compute_amount_payable',
+        help="Final regiter payment amount even after advance clearing",
+    )
+
+    @api.multi
+    @api.depends('expense_line_ids')
+    def _compute_advance(self):
+        for sheet in self:
+            if True in sheet.expense_line_ids.mapped('advance') and \
```

Comment:
```suggestion
            if True in sheet.expense_line_ids.mapped('advance') and \
            sheet.advance = bool(sheet.expense_line_ids.mapped('advance') and len(sheet.expense_line_ids) == 1)
```

Reply:
Thank you for your thoughtful comment. I have changed all of them.